e2etest/l2tests/l2.go

```diff
@@ -464,6 +468,12 @@ var _ = ginkgo.Describe("L2", func() {
 			}, 2*time.Minute, 1*time.Second).Should(gomega.BeNil())
 
 			// Negative - validate that the other speakers don't publish layer2 metrics
+			advNode, err := advertisingNodeFromMAC(allNodes.Items, ingressIP, executor.Host)
```

Comment:
You can avoid repeating the last round of the loop by having advNode and advSpeaker outside and not inited inside the eventually.

Reply:
As far as I know and tried I can't do that because they are set inside a function in the eventually.